### DISCUSSION TITLE
How to get the coordinates of points along a line ?

### DISCUSSION THREAD

**Discussion by Antoine-S-C** [cwapi3d/cwapi3dpython#225] [General]
Hello everybody,
1- How to get the coordinates of n points distributed along a polyline or spline?
2- How to get the coordinates of points separated by a distance (given by the user) along a polyline or spline?

**Comment by Antoine-S-C** [comment]
And similar question: how to get the position of all points of a polyline?
I can only get the first and second using the functions "gc.get_p1(elem)" and "gc.get_p2(elem)"
@Brunner246

**Reply by Brunner246** [reply]
1. it is just possible to get the begin and end vertice of a line segment (polyline)

<img src="https://github.com/user-attachments/assets/a77554c3-d657-4204-a1e4-56f277ae31cf" width="160">

```python
import cadwork
import element_controller as ec
import geometry_controller as gc

ec.get_active_identifiable_element_ids()
[1547745100]
gc.get_element_vertices(1547745100)
[[-15615.055782, 14339.065342, 3861.642972], [-15400.055782, 14339.065342, 3861.642971], [-15400.055741, 14875.151636, 2500.000000]]
```
2. there isnt any api function provided for that use case

**Comment by Antoine-S-C** [comment]
Thank you, this will be useful to me.

Splines :
It is possible to divide a spline using the menu : "Auxiliary lines" / "Divide".
I am trying to get the same points in python. Is it not possible?

**Reply by Brunner246** [reply]
This is not possible.
Currently this is only possible via Rhino.Inside cadwork.
